controllers/auth/socials/createUserFromSocial.js

```diff
@@ -0,0 +1,37 @@
+import models from "../../../models";
+
+const { User } = models;
+/**
+ * @description create a user from social authentication profile
+ * @param  {object} profile - user data from social website (eg : twitter, google , or facebook)
+ * @return {User} - a user from the database
+ */
+const createUserFromSocial = async profile => {
+  let email;
+  if (profile.emails) {
+    email = profile.emails[0].value;
+  } else {
+    email = `${profile.id}@ah.com`;
+  }
+  try {
+    const user = await User.findOrCreate({
+      where: {
+        $or: [{ socialId: profile.id }, { email }]
+      },
+      defaults: {
+        username:
+          profile.username || `${profile.name.familyName}_${profile.name.givenName}`,
```

Comment:
what if the username already exists in the database? what do you think about adding a random string or something like a timestamp to make the new username really unique?

Reply:
Let me apply the idea of timestamp to make the username unique...